.travis.yml

```diff
@@ -68,7 +68,8 @@ script:
    }
    JavaApiTest() {
      pushd test/java
-     local status=${ant}
+     ant
+     local status=$?
```

Comment:
You should not rely on $? in Travis (https://docs.travis-ci.com/user/customizing-the-build#Note-on-%24%3F). Just use $(ant), which was probably what you wanted initially?

Reply:
`$(ant)` will capture the full text output from Ant, excluding it from the job log. This is _not_ what I want.

`JavaApiTest()` needs to return the _exit code_ from `ant`, but also needs to `popd` before returning.

Perhaps:
```bash
local status=0
ant || status=1
```

?

Reply:
Yes, that looks good.

Reply:
✅